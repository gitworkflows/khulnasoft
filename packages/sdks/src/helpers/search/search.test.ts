import { convertSearchParamsToQueryObject } from './search';

const querystring =
  'someotherValue=jklsjfdal&abc=klfdjklgfds&khulnasoft.cachebust=true&khulnasoft.preview=page&khulnasoft.noCache=true&__khulnasoft_editing__=true&khulnasoft.overrides.page=037948e52eaf4743afed464f02c70da4&khulnasoft.overrides.037948e52eaf4743afed464f02c70da4=037948e52eaf4743afed464f02c70da4&khulnasoft.overrides.page%3A%2F=037948e52eaf4743afed464f02c70da4&preview_theme_id=128854393017';

const url = new URL(`localhost:3000/about-us?${querystring}`);

describe('convertSearchParamsToQueryObject', () => {
  test('correctly converts URLSearchParams to object', () => {
    const output = convertSearchParamsToQueryObject(url.searchParams);
    expect(output).toMatchSnapshot();
  });
});
