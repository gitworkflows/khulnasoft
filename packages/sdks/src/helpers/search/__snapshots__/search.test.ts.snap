// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`convertSearchParamsToQueryObject > correctly converts URLSearchParams to object 1`] = `
{
  "__khulnasoft_editing__": "true",
  "abc": "klfdjklgfds",
  "khulnasoft.cachebust": "true",
  "khulnasoft.noCache": "true",
  "khulnasoft.overrides.037948e52eaf4743afed464f02c70da4": "037948e52eaf4743afed464f02c70da4",
  "khulnasoft.overrides.page": "037948e52eaf4743afed464f02c70da4",
  "khulnasoft.overrides.page:/": "037948e52eaf4743afed464f02c70da4",
  "khulnasoft.preview": "page",
  "preview_theme_id": "128854393017",
  "someotherValue": "jklsjfdal",
}
`;
